File: android-smsmms/src/main/java/com/klinker/android/send_message/SmsManagerFactory.java
package com.klinker.android.send_message;

import android.os.Build;
import android.telephony.SmsManager;

public class SmsManagerFactory {

    public static SmsManager createSmsManager(Settings settings) {
        return createSmsManager(settings.getSubscriptionId());
    }

    public static SmsManager createSmsManager(int subscriptionId) {
        if (subscriptionId != Settings.DEFAULT_SUBSCRIPTION_ID &&
                Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1) {
            SmsManager manager = null;

            try {
                manager = SmsManager.getSmsManagerForSubscriptionId(subscriptionId);
            } catch (Exception e) {
                e.printStackTrace();
            }

            if (manager != null) {
                return manager;
            } else {
                return SmsManager.getDefault();
            }
        } else {
            return SmsManager.getDefault();
        }
    }
}

File: android-smsmms/src/main/java/com/klinker/android/send_message/SmsManagerFactory.kt
package com.klinker.android.send_message

import android.telephony.SmsManager

object SmsManagerFactory {

    fun createSmsManager(subscriptionId: Int): SmsManager {
        var manager: SmsManager? = null

        if (subscriptionId != -1) {
            try {
                manager = SmsManager.getSmsManagerForSubscriptionId(subscriptionId)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        return manager ?: SmsManager.getDefault()
    }
}
